packages/pages/src/components/link/types.ts

```diff
@@ -9,16 +13,77 @@ export const LinkType = {
 
 /**
  * Type of link types that might be received from the platform.
+ *
+ * @public
  */
 export type LinkType = typeof LinkType[keyof typeof LinkType];
 
 /**
  * Type for CTA field
  * Note that when coming from the platform the label will always be a string
  * but ReactNode allows for more general usage.
+ *
+ * @public
  */
 export interface CTA {
   link: string;
   label?: React.ReactNode;
   linkType?: LinkType;
 }
+
+/**
+ * Configuration options available for any usages of the Link component.
+ */
+interface LinkConfig
+  extends React.DetailedHTMLProps<
+    React.AnchorHTMLAttributes<HTMLAnchorElement>,
+    HTMLAnchorElement
+  > {
+  obfuscate?: boolean;
+  eventName?: string;
+  conversionDetails?: ConversionDetails | undefined;
+}
+
+/**
+ * The shape of the data passed to {@link Link} when directly passing an HREF to the Link component.
+ *
+ * @public
+ */
+export interface HREFLinkProps extends LinkConfig {
```

Comment:
if we're exporting these interfaces publicly, we probably also want to export `LinkConfig` so people can get the complete type, right?

Reply:
I meant in terms of documentation purposes, it would be more complete. if we were generating doc pages from these with API extractor, it would complain about `LinkConfig` not being exported. but, if we're not doing that then I guess it's not needed for now